src/Engine/Rendering/Renderer.hpp

```diff
@@ -94,19 +94,55 @@ class RA_ENGINE_API Renderer
     };
 
     /**
-     * Picking result
+     * \brief Result of a PickingQuery
+     *
+     * For each selected pixel, store the following indices:
+     *  - `elementId`: Id of the selected face or edge (for mesh and line mesh respectively). Set to
+     *  -1 for point clouds.
+     *  - `vertexId`: On meshes/lines: id of the closest vertex on the selected face/edge. On
+     *  point-clouds: index of the selected vertex
+     *  - `edgeId`: On meshes/lines: id of the closest edge on the selected face/edge. In meshes,
+     *  `edgeid` is defined using the index of the opposite vertex. Set to -1 for point clouds.
+     *
+     *  For performance reasons, there is no duplicate check when filling PickingResult. When
+     *  required, call removeDuplicatedIndices() before processing indices().
+     *
+     *  \see getPickingResults
      */
     struct PickingResult {
-        PickingMode m_mode;            // Picking mode of the query
-        Core::Utils::Index m_roIdx;    // Idx of the picked RO
-        std::vector<int> m_vertexIdx;  // Idx of the picked vertex in the element, i.e. point's idx
-                                       // OR idx in line or triangle
-        std::vector<int> m_elementIdx; // Idx of the element, i.e. triangle for mesh, edge for lines
-                                       // and -1 for points
-        std::vector<int> m_edgeIdx;    // Idx of the opposite vertex in the triangle if mesh
-        // Note: There is exactly one triplet for each querried pixel (thus there can be doublons
-        // coming from pixels). Note: Beware that the same mesh vertex would also be picked for each
-        // of its adjacent triangles.
+        /// Picking mode of the query
+        PickingMode m_mode {Engine::Rendering::Renderer::RO};
+        /// Idx of the picked RO
+        Core::Utils::Index m_roIdx {Core::Utils::Index::Invalid()};
```

Comment:
Is this field public ?

Reply:
The struct had only public fields. I moved one to private, but true, we could move to a class and hide them all.